pkg/shared/ruleset/disak8sstig/rules/274882.go

```diff
@@ -0,0 +1,116 @@
+}
+
+func (r *Rule274882) Name() string {
+	return "The Kubernetes API server must have anonymous authentication disabled."
+}
+
+func (r *Rule274882) Severity() rule.SeverityLevel {
+	return rule.SeverityHigh
+}
+
+func (r *Rule274882) Run(ctx context.Context) (rule.RuleResult, error) {
+
+	const encryptionProviderConfigOption = "encryption-provider-config"
+
+	deploymentName := "kube-apiserver"
+	containerName := "kube-apiserver"
+
+	if r.DeploymentName != "" {
+		deploymentName = r.DeploymentName
+	}
+
+	if r.ContainerName != "" {
+		containerName = r.ContainerName
+	}
+
+	target := rule.NewTarget("name", deploymentName, "namespace", r.Namespace, "kind", "Deployment")
+
+	optSlice, err := kubeutils.GetCommandOptionFromDeployment(ctx, r.Client, deploymentName, containerName, r.Namespace, encryptionProviderConfigOption)
+	if err != nil {
+		return rule.Result(r, rule.ErroredCheckResult(err.Error(), target)), nil
+	}
+
+	if len(optSlice) > 1 {
+		return rule.Result(r, rule.WarningCheckResult(fmt.Sprintf("Option %s has been set more than once in the container command.", encryptionProviderConfigOption), target)), nil
+	} else if len(optSlice) == 0 {
+		return rule.Result(r, rule.FailedCheckResult(fmt.Sprintf("Option %s has not been set in the container command.", encryptionProviderConfigOption), target)), nil
+	}
+
+	kubeAPIServerDeployment := &appsv1.Deployment{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: r.Namespace,
+		},
+	}
+
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(kubeAPIServerDeployment), kubeAPIServerDeployment); err != nil {
+		return rule.Result(r, rule.ErroredCheckResult(err.Error(), target)), nil
+	}
+
+	bytes, err := kubeutils.GetVolumeConfigByteSliceByMountPath(ctx, r.Client, kubeAPIServerDeployment, containerName, optSlice[0])
+	if err != nil {
+		return rule.Result(r, rule.ErroredCheckResult(err.Error(), target)), nil
+	}
+
+	encryptionProviderConfig := &apiserverv1.EncryptionConfiguration{}
+	if err := yaml.Unmarshal(bytes, encryptionProviderConfig); err != nil {
+		return rule.Result(r, rule.ErroredCheckResult(err.Error(), target)), nil
+	}
+
+	if len(encryptionProviderConfig.Resources) == 0 {
+		return rule.Result(r, rule.FailedCheckResult("Secrets are not explicitly encrypted at REST.", target)), nil
+	}
+
+	for _, resourceConfig := range encryptionProviderConfig.Resources {
+		if slices.ContainsFunc(resourceConfig.Resources, func(resourceName string) bool {
+			return resourceName == "*." || resourceName == "*.*" || resourceName == "secrets"
+		}) {
+			if len(resourceConfig.Providers) == 0 {
+				return rule.Result(r, rule.FailedCheckResult("Secrets are not explicitly encrypted at REST.", target)), nil
+			}
+
+			if resourceConfig.Providers[0].Identity != nil {
+				return rule.Result(r, rule.FailedCheckResult("Secrets are explicitly stored as plain text.", target)), nil
+			}
+
+			return rule.Result(r, rule.PassedCheckResult("Secrets are encrypted at REST.", target)), nil
```

Comment:
WDYT about checking the number of set providers here? It should be acceptable to have only 1 set provider.

```suggestion
			setProviders := 0
			if resourceConfig.Providers[0].AESGCM != nil {
				setProviders++
			}
			if resourceConfig.Providers[0].AESCBC != nil {
				setProviders++
			}
			if resourceConfig.Providers[0].Secretbox != nil {
				setProviders++
			}
			if resourceConfig.Providers[0].KMS != nil {
				setProviders++
			}

			switch setProviders {
			case 0:
				return rule.Result(r, rule.FailedCheckResult("No provider has been set for secrets encryption at REST.", target)), nil
			case 1:
				return rule.Result(r, rule.PassedCheckResult("Secrets are encrypted at REST.", target)), nil
			default:
				return rule.Result(r, rule.WarningCheckResult("Multiple encryption providers are set for secrets encryption at REST.", target)), nil
			}
```